mne/annotations.py

```diff
@@ -443,3 +443,108 @@ def _ensure_annotation_object(obj):
     if not isinstance(obj, Annotations):
         raise ValueError('Annotations must be an instance of '
                          'mne.Annotations. Got %s.' % obj)
+
+
+def _strip_to_integer(trigger):
+    """Return only the integer part of a string."""
+    return int("".join([x for x in trigger if x.isdigit()]))
+
+
+@verbose
+def events_from_annotations(raw, event_id=None, regexp=None,
+                            event_id_func=None, verbose=None):
+    """Get events and event_id from an Annotations object.
+
+    Parameters
+    ----------
+    raw : instance of Raw
+        The raw data for which Annotations are defined.
+
+    event_id : dict | None
+        Dictionary of string keys and integer values as used in mne.Epochs
+        to map annotation descriptions to integer event codes. Only the
+        keys present will be mapped and the annotations with other descriptions
+        will be ignored. If None, all descriptions of annotations are mapped
+        and assigned arbitrary unique integer values.
+
+    regexp : str | None
+        Regular expression used to filter the annotations whose
+        descriptions is a match.
+
+    event_id_func : None | str | callable
+        What to do for events not found in ``event_id``. Must take one ``str``
+        argument and return an ``int``. If string, must be 'strip-to-integer',
+        in which case it defaults to stripping event codes such as "D128" or
```

Comment:
@agramfort what is this "D128" / "S  1", are they a *fif* thing or its only eeglab stuff?

Reply:
we actually have no `eeglab` file with such naming convention. All the descriptions are `['rt', 'square']` otherwise this test would break (and it does not):

```py
@pytest.mark.parametrize("fname", raw_fnames)
@testing.requires_testing_data
def test_eeglab_annotations(fname):
    """Test reading annotations in EEGLAB files."""
    _check_h5(fname)
    annotations = read_annotations_eeglab(fname)
    assert len(annotations) == 154
    assert set(annotations.description) == set(['rt', 'square'])
    assert np.all(annotations.duration == 0.)
```

Reply:
The only time that `_strip_to_integer(trigger)` is called is here https://github.com/mne-tools/mne-python/blob/8473339e37dec97c84574ab718b7a61b7f1ce11c/mne/io/eeglab/eeglab.py#L710-L714 and always returns VAlueError in the testsuit. 

```
In [6]: from mne.io.eeglab import _strip_to_integer

In [7]: _strip_to_integer('rt')
Traceback (most recent call last):
  File "<ipython-input-7-5d985d8c35de>", line 1, in <module>
    _strip_to_integer('rt')
  File "/home/sik/code/mne-python/mne/io/eeglab/eeglab.py", line 824, in _strip_to_integer
    return int("".join([x for x in trigger if x.isdigit()]))
ValueError: invalid literal for int() with base 10: ''


In [8]: _strip_to_integer('square')
Traceback (most recent call last):
  File "<ipython-input-8-4a7ea79a8b44>", line 1, in <module>
    _strip_to_integer('square')
  File "/home/sik/code/mne-python/mne/io/eeglab/eeglab.py", line 824, in _strip_to_integer
    return int("".join([x for x in trigger if x.isdigit()]))
ValueError: invalid literal for int() with base 10: ''


In [9]: 
```

Does anybody use it ?